.github/workflows/macOS-CMake-toolchain.yml

```diff
@@ -0,0 +1,88 @@
+name: macOS (CMake toolchain)
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: macos-latest
+
+    env:
+      # Hardcoded date selecting the latest toolchain from https://swift.org/download/#snapshots.
+      toolchain-date: 2021-01-04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set environment variables
+        run: |
+          TOOLCHAIN_DATE=${{ env.toolchain-date }}
+          # TOOLCHAIN_SHORT_DATE: TOOLCHAIN_DATE with hypens removed. Used for toolchain identifier and version.
+          TOOLCHAIN_SHORT_DATE="${TOOLCHAIN_DATE//-/}"
+          TOOLCHAIN_NAME=swift-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a
+          TF_TOOLCHAIN_NAME=swift-tensorflow-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a
+          echo "TOOLCHAIN_DATE=${TOOLCHAIN_DATE}" >> $GITHUB_ENV
+          echo "TOOLCHAIN_SHORT_DATE=${TOOLCHAIN_SHORT_DATE}" >> $GITHUB_ENV
+          echo "TOOLCHAIN_NAME=${TOOLCHAIN_NAME}" >> $GITHUB_ENV
+          echo "TF_TOOLCHAIN_NAME=${TF_TOOLCHAIN_NAME}" >> $GITHUB_ENV
+          echo "SWIFT_APIS_BUILD_DIR=/tmp/BinaryCache/tensorflow-swift-apis" >> $GITHUB_ENV
+          echo "TOOLCHAIN_ROOT_LOCATION=/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain" >> $GITHUB_ENV
+          echo "TOOLCHAIN_HOME_LOCATION=$HOME/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain" >> $GITHUB_ENV
+
+      - name: Install Ninja
+        run: |
+          brew install ninja
+
+      - name: Install swift.org toolchain
+        run: |
+          curl -sOL https://swift.org/builds/development/xcode/${TOOLCHAIN_NAME}/${TOOLCHAIN_NAME}-osx.pkg
+          xattr -dr com.apple.quarantine ${TOOLCHAIN_NAME}-osx.pkg
+          installer -pkg ${TOOLCHAIN_NAME}-osx.pkg -target CurrentUserHomeDirectory
+          rm -rf ${TOOLCHAIN_NAME}-osx.pkg
+
+      - name: Install pre-built TensorFlow and X10 libraries
+        run: |
+          curl -sL https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80NTU3NC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy1kYXJ3aW4teDY00/content?format=zip -o tensorflow-darwin-x64.zip
+          unzip tensorflow-darwin-x64.zip
+          mv tensorflow-darwin-x64/Library/tensorflow-2.4.0 ~/Library/
+
+      - name: Build tensorflow/swift-apis via CMake
+        run: |
+          set -euxo pipefail
+          mkdir -p $SWIFT_APIS_BUILD_DIR
+          cmake \
+            -B $SWIFT_APIS_BUILD_DIR \
+            -D BUILD_SHARED_LIBS=YES \
+            -D CMAKE_BUILD_TYPE=Release \
+            -D CMAKE_INSTALL_PREFIX=${TOOLCHAIN_HOME_LOCATION}/usr \
+            -D CMAKE_Swift_COMPILER=${TOOLCHAIN_HOME_LOCATION}/usr/bin/swiftc \
```

Comment:
I think it would be nicer to use `TOOLCHAINS` and `xcrun` to get the path to `swiftc` a la:

```
-D CMAKE_Swift_COMPILER=$(TOOLCHAINS=org.swift.50202101041a xcrun -f swiftc)
```

Reply:
Sounds good to me. I'm curious why you prefer calling `xcrun -f swiftc` when the exact path to `swiftc` is already known?

Reply:
I guess the advantage of using `TOOLCHAINS=... xcrun -f swiftc` is that it finds where `swiftc` is installed, and is robust against toolchain install location.

However, I actually think it is more portable to not use `xcrun` so that `macOS-CMake-toolchain.yml` can be more easily adapted to other platforms.

Reply:
I dont think that this action is portable at all.  `pkg` is very much a macOS operation.  The installation process, the build, etc are platform dependent.  We could make the rule more sharable by relying on the PATH.  However, all that makes sharable is a single line, which IMO doesn't add much value.

Reply:
Good points about non-portability, thanks!

Reply:
Done.

Reply:
Using `TOOLCHAINS=... xcrun -f swiftc` doesn't work when the toolchain installer is unpacked twice: once as a source toolchain for compiling `tensorflow/swift-apis`, and second as a destination toolchain for installing `tensorflow/swift-apis` and repackaging.

The two toolchains share the same `com.swift.50202101041a` identifier, so `xcrun` can't resolve to the `swiftc` path that we want (the destination toolchain one).

<details>
<p>

```
  TOOLCHAINS=org.swift.50${TOOLCHAIN_SHORT_DATE}a
  SOURCE_TOOLCHAIN_SWIFTC=$(TOOLCHAINS=${TOOLCHAINS} xcrun -f swiftc)
  if [[ ${SOURCE_TOOLCHAIN_SWIFTC} == *"Xcode"* ]]; then
    echo "Unexpected Xcode toolchain swiftc found; incorrect TOOLCHAINS identifier."
    exit 1
  fi
  cmake \
    -B build \
    -D BUILD_SHARED_LIBS=YES \
    -D CMAKE_BUILD_TYPE=Release \
    -D CMAKE_INSTALL_PREFIX=${DEST_TOOLCHAIN_LOCATION}/usr \
    -D CMAKE_Swift_COMPILER=${SOURCE_TOOLCHAIN_SWIFTC} \
    -D CMAKE_OSX_DEPLOYMENT_TARGET=10.9 \
    -D TENSORFLOW_USE_STANDARD_TOOLCHAIN=YES \
    -D X10_LIBRARY=${HOME}/Library/tensorflow-2.4.0/usr/lib/libx10.dylib \
    -D X10_INCLUDE_DIRS=${HOME}/Library/tensorflow-2.4.0/usr/include \
    -D BUILD_TESTING=NO \
    -G Ninja \
    -S ${PWD}
  shell: /bin/bash -e {0}
  env:
    toolchain-date: 2021-01-04
    TOOLCHAIN_DATE: 2021-01-04
    TOOLCHAIN_SHORT_DATE: 20210104
    TOOLCHAIN_NAME: swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a
    SOURCE_TOOLCHAIN_LOCATION: /Users/runner/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain
    DEST_TOOLCHAIN_LOCATION: /tmp/destination-toolchain/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain
2021-01-07 03:07:17.443 xcodebuild[1284:9049] [MT] DVTToolchain: Failed to register toolchain: <DVTToolchain:0x7f8b0c45eb30:'org.swift.50202101041a':'Swift Development Snapshot 2021-01-04 (a)'>: Error Domain=DVTFoundationErrorDomain Code=-1 "Can't register <DVTFilePath:0x7f8b0c459020:'/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>; identifier org.swift.50202101041a conflicts with <DVTFilePath:0x7f8b0c459190:'/Users/runner/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>" UserInfo={NSLocalizedDescription=Can't register <DVTFilePath:0x7f8b0c459020:'/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>; identifier org.swift.50202101041a conflicts with <DVTFilePath:0x7f8b0c459190:'/Users/runner/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>}
2021-01-07 03:07:18.964 xcodebuild[1286:9170] [MT] DVTToolchain: Failed to register toolchain: <DVTToolchain:0x7fe214748ba0:'org.swift.50202101041a':'Swift Development Snapshot 2021-01-04 (a)'>: Error Domain=DVTFoundationErrorDomain Code=-1 "Can't register <DVTFilePath:0x7fe2147433d0:'/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>; identifier org.swift.50202101041a conflicts with <DVTFilePath:0x7fe214742c80:'/Users/runner/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>" UserInfo={NSLocalizedDescription=Can't register <DVTFilePath:0x7fe2147433d0:'/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>; identifier org.swift.50202101041a conflicts with <DVTFilePath:0x7fe214742c80:'/Users/runner/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-04-a.xctoolchain'>}
Unexpected Xcode toolchain swiftc found; incorrect TOOLCHAINS identifier.
```

</p>

</details>

Reverted to hardcoding the destination toolchain `swiftc` path. This seems more direct and straightforward.